code/datums/station_state.dm

```diff
@@ -0,0 +1,76 @@
+// This used to be part of the blob gamemode files. Why.
+/datum/station_state
+	/// Amount of floors
+	var/floor = 0
+	/// Amount of regular walls
+	var/wall = 0
+	/// Amount of reinforced walles
+	var/r_wall = 0
+	/// Amount of full windows
+	var/window = 0
+	/// Amount of airlocks
+	var/door = 0
+	/// Amount of grilles (without windows)
+	var/grille = 0
+	/// Amount of machines
+	var/mach = 0
+
+/datum/station_state/proc/count()
+	var/watch = start_watch()
+	log_debug("Counting station atoms")
+	var/station_zlevel = level_name_to_num(MAIN_STATION)
+	for(var/turf/T in block(locate(1, 1, station_zlevel), locate(world.maxx, world.maxy, station_zlevel)))
+
+		if(istype(T, /turf/simulated/floor))
+			if(!(T:burnt))
```

Comment:
wouldn't it be better to type cast this and then use the `.` operator?

Reply:
Im divided on this. On the one hand, its cleaner code, on the other hand, this needs to run quickly and I would need to measure how much the extra typecast is, since we already do a heavy amount of it on line 22.